File: framework/petscFinalized.m
function [finalized, errCode, toplevel] = petscFinalized
%Determine whether PetscFinalize() has been called yet.
%   [finalized, errCode] = petscFinalized
%
%   finalized(int)
%   errCode(int) return code (0 indicates OK)
%
% SEE ALSO: petscInitialize, petscFinalize, petscInitialized
%
% PETSc C interface:
%   PetscErrorCode  PetscFinalized(PetscBool  *isFinalized)
% http://www.mcs.anl.gov/petsc/petsc-current/docs/manualpages/Sys/PetscFinalized.html

%#codegen -args {}

finalized = int32(0);
errCode = int32(-1);

if ~coder.target('MATLAB')
    b = coder.opaque('PetscBool');
    errCode = coder.ceval('PetscFinalized', coder.wref(b));
    
    toplevel = nargout>2;
    if errCode && (toplevel || m2c_debug)
        m2c_error('petsc:RuntimeError', 'PetscFinalized returned error code %d\n', errCode)
    end

    finalized = coder.ceval(' ', b);
end
end

File: sys/codegen/lib/petscFinalized/petscFinalized.c
#include "petscFinalized.h"
#include "m2c.h"
#include "petsc4m.h"

static void m2c_error(int varargin_3);

static void m2c_error(int varargin_3)
{
  M2C_error("petsc:RuntimeError", "PetscFinalized returned error code %d\n",
            varargin_3);
}

void petscFinalized(int *finalized, int *errCode, boolean_T *toplevel)
{
  PetscBool b;
  *errCode = PetscFinalized(&b);
  *toplevel = true;
  if (*errCode != 0) {
    m2c_error(*errCode);
  }
  *finalized = (b);
}

void petscFinalized_initialize(void)
{
}

void petscFinalized_terminate(void)
{
}
